openid-4-verifiable-presentations-1_0.md

```diff
@@ -1410,6 +1410,22 @@ into the key derivation process that is used to derive the content encryption ke
 
 Note: For encryption, implementers have a variety of options available through JOSE, including the use of Hybrid Public Key Encryption (HPKE) as detailed in [@I-D.ietf-jose-hpke-encrypt]. 
 
+### ECDH-ES APU/APV Values
+When performing Reponse Encryption where the `alg` is ECDH-ES the `apv` MUST be set by the Wallet and validated by the Verifier. The value to set is is the base64url encodig of the sha-256 hash of the bytes of several fields concatanated. The fields to be concatanated is based on the response mode:
+
+  - When response mode is `direct_post.jwt` the clientId, nonce, jwkThumbprint and responseUri
+  - When response mode is `dc_api.jwt` the origin, nonce and jwkThumbprint
```

Comment:
Per https://www.rfc-editor.org/rfc/rfc7638.html#section-3, the JWK Thumbprint is the hash of the bytes of a canonical representation of the JWK.  When SHA-256 is the hash function, this is 32 bytes of binary data.  Is that what you intend to include in the concatenation?  Or do you intend to include the base64url-encoding of the JWK Thumbprint in the concatenation?